commit 9f497bcc695fb828da023d74ad3c966b1e58ad21
Author: Milan Broz <mbroz@redhat.com>
Date:   Tue Aug 12 17:48:27 2008 +0200

    ACPI: Fix thermal shutdowns
    
    Do not use unsigned int if there is test for negative number...
    
    See drivers/acpi/processor_perflib.c
      static unsigned int ignore_ppc = -1;
    ...
      if (event == CPUFREQ_START && ignore_ppc <= 0) {
           ignore_ppc = 0;
    ...
    
    Signed-off-by: Milan Broz <mbroz@redhat.com>
    Signed-off-by: Andi Kleen <ak@linux.intel.com>

diff --git a/drivers/acpi/processor_perflib.c b/drivers/acpi/processor_perflib.c
index 0133af4..80e3209 100644
--- a/drivers/acpi/processor_perflib.c
+++ b/drivers/acpi/processor_perflib.c
@@ -70,7 +70,7 @@ static DEFINE_MUTEX(performance_mutex);
  *  0 -> cpufreq low level drivers initialized -> consider _PPC values
  *  1 -> ignore _PPC totally -> forced by user through boot param
  */
-static unsigned int ignore_ppc = -1;
+static int ignore_ppc = -1;
 module_param(ignore_ppc, uint, 0644);
 MODULE_PARM_DESC(ignore_ppc, "If the frequency of your machine gets wrongly" \
 		 "limited by BIOS, this should help");
